core/src/main/scala/api/sfs.scala

```diff
@@ -25,15 +30,19 @@ trait Filesystem {
    *  things too eagerly, nor caching data for too long.
    */
   def resolve(path: Path): Key
-  def metadata(key: Key): Metadata
-  def lookup(key: Key): Data
+  def metadata(key: Key): M[Metadata]
+  def lookup(key: Key): M[Data]
```

Comment:
> Are you proposing we block on that?

No, I'm proposing we block on absolutely nothing - I'm embracing the same notion the rest of the world is embracing. You return a metadata map immediately, blocking on nothing. Maybe it has `Size` because the bytes are local and fixed, maybe it has `Eval[Size]` because the file size must be computed, maybe it has `Future[Size]` because the size must be queried over the network. In all cases you get the metadata immediately, but the amount of indirection remaining varies situationally.